File: 2657-find-the-prefix-common-array-of-two-arrays/main.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
                
        n = len(A)
        ans = []
        set_a, set_b = set(), set()

        for idx in range(n):
            set_a.add(A[idx])
            set_b.add(B[idx])

            common = len(set_a.intersection(set_b))
            ans.append(common)
        
        return ans

File: 2657-find-the-prefix-common-array-of-two-arrays/main.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        
        int n = A.size();
        vector<int> result(n);
        unordered_set<int> setA, setB;

        for (int i = 0; i < n; i++) {
            setA.insert(A[i]);
            setB.insert(B[i]);

            int common = 0;
            for (const auto& num: setA) {
                if (setB.count(num)) common++;
            }

            result[i] = common;
        }

        return result;
    }
};
